src/TRestGeant4ToDetectorHitsProcess.cxx

```diff
@@ -207,12 +207,15 @@ TRestEvent* TRestGeant4ToDetectorHitsProcess::ProcessEvent(TRestEvent* evInput)
         const auto& track = fG4Event->GetTrack(i);
         const auto& hits = track.GetHits();
         for (int j = 0; j < track.GetNumberOfHits(); j++) {
+            const auto energy = hits.GetEnergy(j);
             for (const auto& volumeID : fVolumeId) {
-                const auto energy = hits.GetEnergy(j);
-                if ((hits.GetVolumeId(j) == volumeID) && (energy > 0)) {
+                if (hits.GetVolumeId(j) == volumeID && energy > 0) {
                     fHitsEvent->AddHit(hits.GetX(j), hits.GetY(j), hits.GetZ(j), energy);
                 }
             }
+            if (fVolumeId.empty() && energy > 0) {
```

Comment:
@lobis But how it can happen that `fVolumeId` is empty?

Reply:
Or making the question in a different way. To which volume do those hits belong to?